centml/compiler/server_compilation.py

```diff
@@ -32,21 +74,26 @@ def dir_cleanup(model_id: str):
         raise Exception("Failed to delete the directory") from e
 
 
-def hidet_backend_server(graph_module: GraphModule, example_inputs: List[Tensor], model_id: str):
-    assert isinstance(graph_module, GraphModule)
+def hidet_backend_server(input_graph_module: GraphModule, example_inputs: List[torch.Tensor], model_id: str):
+    assert isinstance(input_graph_module, GraphModule)
 
-    logger.info("received a subgraph with %d nodes to optimize", len(graph_module.graph.nodes))
-    logger.debug("graph: %s", graph_module.graph)
-
-    interpreter: Interpreter = from_torch(graph_module)
-    flow_graph, _, _ = get_flow_graph(interpreter, example_inputs)
+    # Create hidet compiled graph
+    interpreter: Interpreter = from_torch(input_graph_module)
+    flow_graph, _, output_format = get_flow_graph(interpreter, example_inputs)
     cgraph = get_compiled_graph(flow_graph)
 
-    # perform inference using example inputs to get dispatch table
+    # Perform inference using example inputs to get dispatch table
     hidet_inputs = preprocess_inputs(example_inputs)
     cgraph.run_async(hidet_inputs)
 
+    # Get compiled forward function
+    wrapper = HidetCompiledModel(cgraph, hidet_inputs, output_format)
+
+    # Wrap the forward function in a torch.fx.GraphModule
+    compiled_graph_module = get_graph_module(wrapper)
```

Comment:
do we know if the runtime of this wrapped module is as good as running the HidetCompiledModel directly?

Reply:
yup! I verified by comparing the inference times of compiling with centml backend vs just hidet backend over ~1000 iterations each for:
- resnet18
- bert
- gpt2 (using model.generate for inference)

In all cases the inference times were basically the same (their variation was explainable by randomness)